core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -54,22 +59,43 @@ private Fuzziness(int fuzziness) {
     }
 
     private Fuzziness(String fuzziness) {
-        if (fuzziness == null) {
+        if (fuzziness == null || fuzziness.isEmpty()) {
             throw new IllegalArgumentException("fuzziness can't be null!");
         }
         this.fuzziness = fuzziness.toUpperCase(Locale.ROOT);
     }
 
+    private Fuzziness(String fuzziness, int lowDistance, int highDistance) {
+        this(fuzziness);
+        if (lowDistance < 0 || highDistance < 0 || lowDistance > highDistance) {
+            throw new IllegalArgumentException("fuzziness wrongly configured, must be: lowDistance > 0, highDistance" +
+                " > 0 and lowDistance <= highDistance ");
+        }
+        this.lowDistance = lowDistance;
+        this.highDistance = highDistance;
+    }
+
     /**
      * Read from a stream.
      */
     public Fuzziness(StreamInput in) throws IOException {
         fuzziness = in.readString();
+        if (in.readBoolean()) {
+            lowDistance = in.readVInt();
+            highDistance = in.readVInt();
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(fuzziness);
+        if (isAutoWithCustomValues()) {
```

Comment:
similarly, older versions do not know about these new options, so you need to do something like this:

```
if (out.getVersion().before(Version.V_6_1_0)) {
    // we cannot serialize the low/high bounds since the other node does not know about them
    // so we serialize Fuzziness.AUTO instead. This is a best-effort to not fail queries in case
    // the cluster is being upgraded and users start using features that are not available on all
    // nodes.
    Fuzziness.AUTO.writeTo(out);
} else {
    out.writeString(fuzziness);
    if (isAutoWithCustomValues()) {
    [...]
}
```